applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -562,18 +612,37 @@ static void discover_source_cb(struct bt_conn *conn, struct bt_codec *codec, str
 	}
 
 	if (codec != NULL) {
```

Comment:
This is the way these callbacks (discover_sink_cb and discover_source_cb) are currently designed.
The first time(s) discover_sink/source_cb() is called, "codec != NULL", and we store the codec capabilities.
After that "codec == NULL", and that means we should proceed further in the callback.

This way of doing it will most likely be changed in host in the future